drivers/net/bnxt/bnxt_rxq.c

```diff
@@ -122,8 +122,14 @@ int bnxt_mq_rx_configure(struct bnxt *bp)
 		/* For each pool, allocate MACVLAN CFA rule & VNIC */
 		if (!pools) {
 			RTE_LOG(ERR, PMD,
-				"VMDq pool not set, defaulted to 64\n");
-			pools = ETH_64_POOLS;
+			    "VMDq pool not set, defaulted to %d\n",
+			    RTE_MIN(bp->max_vnics,
+			     RTE_MIN(bp->max_l2_ctx,
+			      RTE_MIN(bp->max_rsscos_ctx, ETH_64_POOLS))));
+			pools = RTE_MIN(bp->max_vnics,
+			    RTE_MIN(bp->max_vnics,
+			     RTE_MIN(bp->max_l2_ctx,
+			      RTE_MIN(bp->max_rsscos_ctx, ETH_64_POOLS))));
```

Comment:
Should we check against max_rsscos_ctx only when RSS flag is set?
Also do we need to export the right max_vmdq_pools/rings in dev_get_info?